.github/workflows/ci.yml

```diff
@@ -15,30 +15,250 @@ jobs:
+    # Only run benchmarks on main branch pushes and PRs to main to avoid excessive CI time
+    if: github.ref == 'refs/heads/main' || github.base_ref == 'main'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+
+      - name: Cache Gradle dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Make gradlew executable
+        run: chmod +x ./gradlew
+
+      - name: Create benchmark results directory
+        run: mkdir -p benchmark-results
+
+      - name: Run serialization benchmarks
+        run: |
+          ./gradlew jmhRunSerializationBenchmarks
+        continue-on-error: true
+
+      - name: Run deserialization benchmarks
+        run: |
+          ./gradlew jmhRunDeserializationBenchmarks
+        continue-on-error: true
+
+      - name: Run field access benchmarks
+        run: |
+          ./gradlew jmhRunFieldAccessBenchmarks
+        continue-on-error: true
+
+      - name: Run size comparison benchmarks
+        run: |
+          ./gradlew jmhRunSizeComparisonBenchmarks
+        continue-on-error: true
+
+      - name: Upload benchmark results
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: benchmark-results-${{ github.sha }}
+          path: benchmark-results/
+          retention-days: 30
+
+      - name: Comment benchmark results on PR
+        if: github.event_name == 'pull_request'
+        uses: actions/github-script@v7
+        continue-on-error: true
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            try {
+              const fs = require('fs');
+              const path = require('path');
+            
+              // Find the latest benchmark results file
+              const resultsDir = 'benchmark-results';
+              let latestFile = null;
+              let latestTime = 0;
+            
+              if (fs.existsSync(resultsDir)) {
+                const files = fs.readdirSync(resultsDir);
+                for (const file of files) {
+                  if (file.endsWith('.json')) {
+                    const filePath = path.join(resultsDir, file);
+                    const stats = fs.statSync(filePath);
+                    if (stats.mtime.getTime() > latestTime) {
+                      latestTime = stats.mtime.getTime();
+                      latestFile = filePath;
+                    }
+                  }
+                }
+              }
+            
+              if (latestFile) {
+                console.log(`📊 Found benchmark results: ${latestFile}`);
```

Comment:
I recommend having the minimal amount of scripting inside `ci.yml` and instead delegate it to a script that we can just run locally. See https://github.com/imprint-serde/imprint/blob/main/scripts/ci_bench.sh for an example, I generate the markdown within the script there and just call that script from the GHA workflow